src/olympia/shelves/forms.py

```diff
@@ -0,0 +1,58 @@
+import requests
+
+from rest_framework.reverse import reverse as drf_reverse
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from olympia.shelves.models import Shelf
+
+
+class ShelfForm(forms.ModelForm):
+    class Meta:
+        model = Shelf
+        fields = ('title', 'shelf_type', 'criteria',
+                  'footer_text', 'footer_pathname',)
+
+    def clean_criteria(self):
+        data = self.cleaned_data
+        criteria = data['criteria']
+        baseUrl = 'https://addons.mozilla.org'
```

Comment:
This certainly shouldn't be a hardcoded url to the production amo site.  If using `settings.EXTERNAL_URL` is causing an error perhaps it's set to an unexpected value?  (either use python debugger or throw in some temporary `print()` statements to see what's going on)